src/compiler/utilities.ts

```diff
@@ -2685,7 +2685,7 @@ export function isCommonJsExportPropertyAssignment(node: Node) {
 export function isValidESSymbolDeclaration(node: Node): boolean {
     return (isVariableDeclaration(node) ? isVarConst(node) && isIdentifier(node.name) && isVariableDeclarationInVariableStatement(node) :
         isPropertyDeclaration(node) ? hasEffectiveReadonlyModifier(node) && hasStaticModifier(node) :
-        isPropertySignature(node) && hasEffectiveReadonlyModifier(node)) || isCommonJsExportPropertyAssignment(node);
+        isPropertySignature(node) && hasEffectiveReadonlyModifier(node)) || isCommonJsExportPropertyAssignment(node) || isJSDocTypeExpression(node);
 }

```

Comment:
can you add a compiler test from #56800:

```js
/** @type {unique symbol} */
const foo = Symbol()
/** @typedef {{ [foo]: boolean }} Alias */
/** @typedef {{ [key: foo] boolean }} BadAlias */
```

The first typedef should work and the second should error, with the same error as Typescript has.